elasticsearch/shard_migration.py

```diff
@@ -0,0 +1,181 @@
+        "-T",
+        "--target-nodes",
+        dest="target_nodes",
+        type="string",
+        help="Comma-separated list of target node names to move shards to.",
+    )
+
+    return parser.parse_args()
+
+
+# Fetch list of shards on the source node
+def get_indices_on_node(node_name, es_host):
+    try:
+        LOG.info(f"Fetching shards from source node: {node_name}.")
+        response = requests.get(f"{es_host}/_cat/shards?v&format=json")
+        response.raise_for_status()
+
+        indices_on_node = []
+        shards_data = response.json()
+
+        # Filter shards by node
+        for shard in shards_data:
+            if shard["node"] == node_name:
+                indices_on_node.append(
+                    {"index": shard["index"], "shard": int(shard["shard"])}
+                )
+
+        LOG.info(f"Found {len(indices_on_node)} shards on source node {node_name}.")
+        return indices_on_node
+
+    except requests.exceptions.RequestException as e:
+        LOG.error(f"Error fetching shards from source node {node_name}: {e}")
+        return []
+
+# Move all shards for an index from a source node to one of the target nodes
+def move_shards(indices, source_node, target_nodes, es_host):
+    for index_data in indices:
+        index = index_data["index"]
+        shard = index_data["shard"]
+
+        for target_node in target_nodes:
+            reroute_body = {
+                "commands": [
+                    {
+                        "move": {
+                            "index": index,
+                            "shard": shard,
+                            "from_node": source_node,
+                            "to_node": target_node,
+                        }
+                    }
+                ]
+            }
+
+            try:
+                LOG.info(
+                    f"Trying to move shard {shard} of index '{index}' from {source_node} to {target_node}."
+                )
+                headers = {"Content-Type": "application/json"}
+                response = requests.post(
+                    f"{es_host}/_cluster/reroute",
+                    headers=headers,
+                    data=json.dumps(reroute_body),
+                )
+                response.raise_for_status()
+                LOG.info(
+                    f"Shard {shard} of index '{index}' successfully moved from {source_node} to {target_node}."
+                )
+                return True
+            except requests.exceptions.RequestException as e:
+                LOG.warning(
+                    f"Not able to move shard {shard} of index '{index}' to {target_node}. Error: {e}"
+                )
+
+        LOG.warning(
+            f"Shard {shard} of index '{index}' could not be moved to any of the target nodes in the list."
+        )
+
+
+def filter_new_shard_allocation(source_node_ip, es_host):
+    try:
+        LOG.info(f"Filtering new shard allocation to on source node: {source_node_ip}.")
```

Comment:
The name of the setting uses the word "exclude", so we should too:
```suggestion
        LOG.info(f"Excluding IP from shard allocation: {source_node_ip}")
```
Also, as always, less is more.